code/modules/awaymissions/zlevel.dm

```diff
@@ -62,6 +62,13 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
 /proc/seedRuins(list/z_levels = null, budget = 0, whitelist = /area/space, list/potentialRuins = space_ruins_templates)
 	if(!z_levels || !z_levels.len)
 		z_levels = list(1)
+
+	for(var/zl in z_levels)
+		var/turf/T = locate(1, 1, zl)
+		if(!T)
+			WARNING("Z level [zl] does not exist - Not generating ruins")
+			return

```

Comment:
@MrStonedOne Neither of the uses of `seedRuins()` ([both here](https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/mapping.dm#L15)) check the array of z-levels for validity. So I could move my changes to `mapping/Initialize()` if you think that's a better place for it, but honestly I think it's fine inside `seedRuins()`